codec/src/test/java/io/netty/handler/codec/MessageAggregatorTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufHolder;
+import io.netty.buffer.DefaultByteBufHolder;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelOutboundHandlerAdapter;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.util.CharsetUtil;
+
+public class MessageAggregatorTest {
+    private static final class ReadCounter extends ChannelOutboundHandlerAdapter {
+        int value;
+
+        @Override
+        public void read(ChannelHandlerContext ctx) throws Exception {
+            value++;
+            ctx.read();
+        }
+    }
+
+    abstract static class MockMessageAggregator
+        extends MessageAggregator<ByteBufHolder, ByteBufHolder, ByteBufHolder, ByteBufHolder> {
+
+        protected MockMessageAggregator() {
+            super(1024);
+        }
+
+        @Override
+        protected ByteBufHolder beginAggregation(ByteBufHolder start, ByteBuf content) throws Exception {
+            return start.replace(content);
+        }
+    }
+
+    private static ByteBufHolder message(String string) {
+        return new DefaultByteBufHolder(
+            Unpooled.copiedBuffer(string.getBytes(CharsetUtil.US_ASCII)));
```

Comment:
nit: `Unpooled.copiedBuffer(string, CharsetUtil.US_ASCII);`

No need to first convert to byte[] and then do the copy